bundles/edu.kit.kastel.sdq.eclipse.grading.view/src/edu/kit/kastel/eclipse/grading/view/controllers/AssessmentViewController.java

```diff
@@ -111,9 +112,32 @@ public void addAssessmentAnnotation(IMistakeType mistake, String customMessage,
 
 	/**
 	 * creates markers for current annotations in the backend
+	 * The method will prevent duplicated markers resulting from 
+	 * having them in the local project and on the server.
 	 */
 	public void createAnnotationsMarkers() {
-		this.getAnnotations().forEach(this::createMarkerForAnnotation);
+		this.getAnnotations().stream().filter(annotation -> !isAnnotationPresent(annotation)).forEach(this::createMarkerForAnnotation);
+	}
+	
+	/**
+	 * Checks whether the given annotation is present in the currently opened project 
+	 * (An annotation is identified by its UUID)
+	 * @param annotation the annotation to check
+	 * @return true if the annotation is present, false if not
+	 */
+	private boolean isAnnotationPresent(IAnnotation annotation) {
+		try {
+			IMarker[] markers = AssessmentUtilities.getFile(annotation.getClassFilePath(), this.systemwideController.getCurrentProjectName()).findMarkers(null, false, 100);
+			for (IMarker marker : markers) {
+				if (annotation.getUUID().equals((String) marker.getAttribute("annotationID"))) {
```

Comment:
Maybe delete this cast. It should not be needed. (Or do I miss sth?)

Reply:
You're right. The way I do it now, it is not required. However, it was required the way I did it before.. :)